src/core/ConversationModel.cpp

```diff
@@ -277,7 +277,11 @@ QVariant ConversationModel::data(const QModelIndex &index, int role) const
     const MessageData &message = messages[index.row()];
 
     switch (role) {
-        case Qt::DisplayRole: return message.text;
+        case Qt::DisplayRole:
+          if (message.text.startsWith(QStringLiteral("/me "))) {
+            return QStringLiteral("* %1 %2").arg(m_contact->nickname(), message.text.mid(4, -1));
```

Comment:
I think that customising text here is OK (better, even, in a lot of ways. The C++ string operations are a lot more straightforward from an operational & performance standpoint than having to do them in JS). Changing later, if it's required, is pretty straightforward.

For customizing the appearance, you could add a new role (IsActionMessageRole, maybe) using the startsWith check for the value, and on the UI side, change the text color or whatever fanciness you find appropriate.